internal/server/proto/server.proto

```diff
@@ -296,6 +296,18 @@ service Waypoint {
   // JSON formatting is not supported.
   rpc WaypointHclFmt(WaypointHclFmtRequest) returns (WaypointHclFmtResponse);
 
+  //----------------------------------------------------------------------
+  // Ondemand runner functions.
+  //----------------------------------------------------------------------
+  rpc UpsertOndemandRunner(UpsertOndemandRunnerRequest) returns (UpsertOndemandRunnerResponse);
+
+  // GetOndemandRunner returns the ondemand runner.
+  rpc GetOndemandRunner(GetOndemandRunnerRequest) returns (GetOndemandRunnerResponse);
```

Comment:
Same with the `Config` suffix on this and List. Its a bit noisy but I think it'll be more idiomatic to our other APIs and if we introduce more ODR APIs will be clearer.